qiskit/passmanager/passmanager.py

```diff
@@ -201,68 +221,143 @@ def run(
         return parallel_map(
-            self._in_parallel, input_programs, task_kwargs={"pm_dill": dill.dumps(self)}
+            _run_workflow_in_new_process,
+            values=in_programs,
+            task_kwargs={"pass_manager_bin": dill.dumps(self)},
         )
 
-    @staticmethod
-    def _in_parallel(
-        in_program: Any,
-        pm_dill: bytes = None,
-    ) -> Any:
-        pass_runner = dill.loads(pm_dill)._create_running_passmanager()
-        return pass_runner.run(in_program)
+    def to_flow_controller(self) -> FlowControllerLinear:
+        """Linearize this manager into a single :class:`.FlowControllerLiner`,
+        so that it can be nested inside another pass manager.
+
+        Returns:
+            A linearized pass manager.
+        """
+        return self._flow_controller
+
+
+def _run_workflow(
+    program: Any,
+    pass_manager: BasePassManager,
+    **kwargs,
+) -> Any:
+    """Run single program optimization with a pass manager.
+
+    Args:
+        program: Arbitrary program to optimize.
+        pass_manager: Pass manager with scheduled passes.
+        **kwargs: Keyword arguments for IR conversion.
+
+    Returns:
+        Optimized program.
+    """
+    flow_controller = pass_manager.to_flow_controller()
+
+    passmanager_ir = pass_manager._passmanager_frontend(
+        input_program=program,
+        **kwargs,
+    )
+    passmanager_ir = flow_controller.execute(
+        passmanager_ir=passmanager_ir,
+        property_set=pass_manager.property_set,
+        callback=kwargs.get("callback", None),
+    )
+    out_program = pass_manager._passmanager_backend(
+        passmanager_ir=passmanager_ir,
+        in_program=program,
+        **kwargs,
+    )
+
+    return out_program
+
+
+def _run_workflow_in_new_process(
+    program: Any,
+    pass_manager_bin: bytes,
+) -> Any:
+    """Run single program optimization in new process.
+
+    Args:
+        program: Arbitrary program to optimize.
+        pass_manager_bin: Binary of the pass manager with scheduled passes.
+
+    Returns:
+          Optimized program.
+    """
+    return _run_workflow(
+        program=program,
+        pass_manager=dill.loads(pass_manager_bin),
+    )
+
+
+def _legacy_build_flow_controller(
+    tasks: list[Task],
+    options: dict[str, Any],
+    **flow_controller_conditions,
+) -> BaseController:
+    """A legacy method to build flow controller with keyword arguments.
+
+    Args:
+        tasks: A list of tasks fed into custom flow controllers.
+        options: Option for flow controllers.
+        flow_controller_conditions: Callables keyed on the alias of the flow controller.
+
+    Returns:
+        A built controller.
+    """
+    warnings.warn(
+        "Building a flow controller with keyword arguments is going to be deprecated. "
+        "Custom controllers must be explicitly instantiated and appended to the task list.",
+        PendingDeprecationWarning,
+    )
```

Comment:
This should set a `stacklevel` that causes it to "blame" the caller of `PassManager.append` (or whichever method) so that it appears to be blamed on user code.  If the `stacklevel` blames library code, then it won't get shown to users at all.  It defaults to 1, which blames this line of code, so that's library code.